### PR TITLE
Connect VM to InteropLib

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Merged in latest from .NET 6 development branch.
Add ComWrappers CMake feature flag
Split interoplibinterface for comwrappers and objc
Create pattern for QCalls

/cc @jkoritzinsky @elinor-fung

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
Can we merge master into the feature branch before reviewing this? There's a ton of unrelated commits right now that make this harder to review.

**Comment by AaronRobinsonMSFT** [comment]
There is 1 commit: https://github.com/dotnet/runtime/pull/47504/commits/9b72a9d06a17a3b4f07c3bad492db2cc84548eb2

**Comment by jkoritzinsky** [comment]
Let's make sure to not squash merge this then.

**Review by jkoritzinsky — APPROVED** [review]
LGTM. This should give us a good starting point to wire through the rest of the way to System.ObjectiveC.Native through interoplib.

File: src/coreclr/vm/interoplibinterface.h [hunk 1]
```diff
@@ -92,6 +95,25 @@ class GlobalComWrappersForTrackerSupport
...
+public:
+    static BOOL QCALLTYPE TrySetGlobalMessageSendCallbacks(
+        void* fptr_objc_msgSend,
+        void* fptr_objc_msgSend_fpret,
+        void* fptr_objc_msgSend_stret,
+        void* fptr_objc_msgSendSuper,
+        void* fptr_objc_msgSendSuper_stret);
+
+    static void QCALLTYPE GetLifetimeMethods(
+            void** allocImpl,
```
Comment (elinor-fung) [inline src/coreclr/vm/interoplibinterface.h:111]: Indentation

**Review by elinor-fung — APPROVED** [review]

**Pull request merged**